docs/how_to/duckdb_register.md

```diff
@@ -0,0 +1,111 @@
+# Use `register` to load external data files with the DuckDB backend
+
+<!-- prettier-ignore-start -->
+Here we use the [`register`][ibis.backends.duckdb.Backend.register] method to load external data files and join them.
+<!-- prettier-ignore-end -->
+
+We're going to download one month of [NYC Taxi
+data](https://www1.nyc.gov/site/tlc/about/tlc-trip-record-data.page) in
+`parquet` format and also download the "Taxi Zone Lookup Table" which is a `csv`
+
+https://d37ci6vzurychx.cloudfront.net/trip-data/green_tripdata_2022-01.parquet
+https://d37ci6vzurychx.cloudfront.net/misc/taxi+_zone_lookup.csv
+
+Create an in-memory DuckDB connection via `ibis`
+
+```python
+>>> import ibis
+>>> con = ibis.duckdb.connect()  # in-memory database
+>>> con.list_tables()
+[]
+```
+
+Now we call `register` with the filepath
+
+```python
+>>> con.register("taxi+_zone_lookup.csv")
+AlchemyTable: taxi+_zone_lookup
+  LocationID   int32
+  Borough      string
+  Zone         string
+  service_zone string
+
+>>> con.register("green_tripdata_2022-01.parquet")
```

Comment:
Do you think showing the use of the optional `table_name` (2nd) argument will show people it's a good way to better control the name of their registered tables? That way if they're iterating through files or want to target specific names for easy reuse they can rather than strictly referencing the table name every time if it's not easily predictable.

e.g. in this case, the hyphen converts to an underscore, which causes the table name to diverge from the file name a bit.

Reply:
seems reasonable